source/common/upstream/health_checker_impl.h

```diff
@@ -360,44 +361,48 @@ class TcpHealthCheckerImpl : public HealthCheckerImplBase {
 
 /**
  * Redis health checker implementation. Sends PING and expects PONG.
+ * TODO(mattklein123): Redis health checking should be via a pluggable module and not in the
+ * "core".
  */
 class RedisHealthCheckerImpl : public HealthCheckerImplBase {
 public:
-  RedisHealthCheckerImpl(const Cluster& cluster, const envoy::api::v2::core::HealthCheck& config,
-                         Event::Dispatcher& dispatcher, Runtime::Loader& runtime,
-                         Runtime::RandomGenerator& random,
-                         Redis::ConnPool::ClientFactory& client_factory);
+  RedisHealthCheckerImpl(
+      const Cluster& cluster, const envoy::api::v2::core::HealthCheck& config,
```

Comment:
This is all fine as-is but in future we may want to to consider typedefs in the class definitions to minimize lots of long namespaces in front of things.  It's totally a style preference but I find the long prefixes everywhere make for less readable code.

Reply:
Sure that's fine. I was just doing find/replace here. No issues with allowing typedefs.